aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -690,21 +692,24 @@ Tensor std_var_common_impl_mps(
                            axes);
 
       num_output_dims = num_input_dims;//(num_input_dims >= num_reduce_dims) ? (num_input_dims - num_reduce_dims) : 0;
-      output_shape = (int64_t *)malloc(num_output_dims  * sizeof(int64_t));
 
       for(int i = 0; i < num_reduce_dims; i++)
       {
-          correction_n *= input_shape[dim_value[i]];
+          auto wrap_dim = maybe_wrap_dim(dim_value[i], input_shape.size());
+          correction_n *= input_shape[wrap_dim];
       }
 
       for (int i = 0; i < num_input_dims; i++)
       {
-          output_shape[i] = [apparent_output_shape[i] longValue];
+          output_shape.push_back([apparent_output_shape[i] longValue]);
       }
   }
 
+  int64_t output_shape_array[output_shape.size()];
+  std::copy(output_shape.begin(), output_shape.end(), output_shape_array);
```

Comment:
Hmm, this uses weird C99 VLA extension, which, to the best of my knowledge is not part of any C++ standard
```suggestion
```